app/models/update_log.rb

```diff
@@ -4,14 +4,31 @@ def self.setting_record
     Setting.find_or_create_by(key: 'metrics_update')
   end
 
-  def self.log_update(time)
+  def self.last_update
+    setting = self.setting_record
+    setting.value.values.last
+
+  end
+
+  def self.log_updates(time)
     setting = self.setting_record
-    setting.value["last_update"] = time
+    last_update = setting.value.keys.last
+    if last_update === nil
+      new_update = 0
+    elsif last_update >= 10
+      setting.value.delete(last_update - 10)
+      new_update = last_update + 1
+    else 
+      new_update = last_update + 1
+    end
+    setting.value[new_update] = time
     setting.save
   end
 
-  def self.last_update
-    self.setting_record
-    Setting.find_by(key: 'metrics_update').value['last_update']
+  def self.average_delay
```

Comment:
This gets called for every page load, so we should calculate it each time the record is updated and save that value on the record itself rather than crunching the numbers again and again.

That will probably require adding another level of nesting to the data structure. So setting.value could be something like:

```
{
  constant_updates: {
    1 => { ... },
    2 => { ... }
   },
  average_constant_update_time: time
}
```